### ISSUE TITLE
Did the field name of filters changed to weights?

### ISSUE THREAD

**Issue by paperplaneuser** [vlfeat/matconvnet#379]
Hi,

First of all, thanks for this great tool.

I was using the code <a href="www.robots.ox.ac.uk/~vgg/practicals/cnn/">here</a>, which usually had net.layers(i).filters as a 4-D array, however now I downloaded MatConvNet 1.0-beta17, and I'm getting errors complaining "no field called 'filters'", and when I check the pretrained models, I see they have weights{1} as a 4-D array. So I wonder if there was a change, and if I should update all my code, so replace "filters" with "weights{1}". Is this true?

Thanks for your time,

<strong>Edit : </strong> I'm also having the same confusion with the field names 'type' and 'class' for layers. For example when I use the cnn_mnist_init.m code to initialize a network and try to get the response from an image, I get the error :

Reference to non-existent field 'class'.

Error in vl_simplenn (line 294)
      res(i+1).x = vl_nnsoftmaxloss(res(i).x, l.class) ;

So should I also change the field name 'type' to 'class' ?

**Comment by vedaldi** [comment]
Hi, preprocess the model using `vl_simplenn_tidy` to upgrade it to the current MatConvNet version.

**Comment by lenck** [comment]
Hi, yes, it has been renamed some versions back...
Just a bit more details what you should probably don. The `vl_simplenn_tidy` would fix your old network and would rename the `filters` and `biases` to `weights`. In this way it would get digested by the `vl_simplenn` and the example function :) This was to clean it up a bit as some functions in the future may need more parameters.

However, if in your code you are accessing the `filters` or `biases`, you would have to replace it with `weights{1}` and `weights{0}` respectively and the `vl_simplenn_tidy` would not help...

With the `class` field - just change the type from `softmaxloss` to `softmax`, that would fix it ;) (`softmaxloss` needs to know the ground truth label, which is stored in the field `class`).
Hop that this help :)

**Comment by paperplaneuser** [comment]
Thanks for both answers. This solved my issue.
